crates/jstzd/src/task/octez_node.rs

```diff
@@ -0,0 +1,264 @@
+use super::Task;
+use anyhow::Result;
+use async_dropper::AsyncDrop;
+use async_trait::async_trait;
+use std::{fs::File, path::PathBuf, sync::Arc};
+use tokio::{runtime::Handle, sync::RwLock};
+
+#[cfg(test)]
+use tests::{Child, InnerOctezNode};
```

Comment:
I think it might be nicer to use `mockall` to Mock these types using `mockall_double` to handle the namespacing issues

Reply:
Additionally I'm not convinced that we want to be mocking everything here. Some things are useful interfaces to mock, but the CI environment should be able to start/stop an octez-node and this should be tested as part of this PR

Reply:
I do think that these are necessary since we definitely don't want each test to spin up a node instance. I've added one integration test to ensure that everything works as expected. This should be sufficient after #567 is merged.

Reply:
I tried to depend on mockall but it is only supported for rust v1.74 and above despite what the doc says, at least the latest few releases of the lib.